sdc_client/balancer.py

```diff
@@ -12,12 +13,51 @@ class StreamsetsBalancer:
     def __init__(self):
         self.streamsets_pipelines: Dict[IStreamSets, List[IPipeline]] = get_streamsets_pipelines()
 
+    def _refresh_object(self):
+        self._preferred_types = self._extract_preferred_types(self.streamsets_pipelines)
+        self._pipelines = self._extract_pipelines(self.streamsets_pipelines)
+        self._streamsets = self._extract_streamsets(self.streamsets_pipelines)
+        self.balanced_streamsets_pipelines: Dict[IStreamSets, List[IPipeline]] = {ss: [] for ss in self._streamsets}
+        self.rebalance_map = {}
+        self.structure = self._get_streamsets_structure(self.streamsets_pipelines)
+
+    def _balance_type(self, type_: str):
+        # Set all typed pipelines to first streamsets with same type
+        pipelines = self.structure[type_]['pipelines']
+        streamsets = self.structure[type_]['streamsets']
+        for pipeline_ in pipelines:
+            self.rebalance_map[pipeline_] = streamsets[0]
+            self.balanced_streamsets_pipelines[streamsets[0]].append(pipeline_)
+        sub_pipelines_dict = {streamsets[0]: pipelines}
+        for ss in streamsets[1:]:
+            sub_pipelines_dict[ss] = []
+        # balance it
+        streamsets_pipelines = sub_pipelines_dict
+        while not self.is_balanced(streamsets_pipelines):
+            streamsets = most_loaded_streamsets(streamsets_pipelines)
+            pipeline = self.streamsets_pipelines[streamsets].pop()
```

Comment:
In this while loop we are working with local `streamsets_pipelines`, but `pipeline = self.streamsets_pipelines[streamsets].pop()` performed from self object, is this OK?

Reply:
In this case it doesn't metter, but mb it was at previos iteration of development.
pop() form local object is better, I changed it